pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/DeletedObjectDao.java

```diff
@@ -0,0 +1,201 @@
+package org.broadinstitute.dsm.db.dao;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.structure.DBElement;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.model.patch.DeleteType;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.lddp.db.SimpleResult;
+
+@Slf4j
+public class DeletedObjectDao {
+
+    private static final String SQL_GET_JSON_OBJECT_FOR_TISSUE = "select concat('select json_object(', group_concat( "
+            + " concat(quote(column_name), ', ', column_name)), ') from ddp_tissue where tissue_id = ? ;') into @sql "
+            + " from information_schema.columns where table_name = 'ddp_tissue'";
+
+    private static final String SQL_GET_JSON_OBJECT_FOR_SM_ID = "select concat('select json_object(', group_concat( "
+            + " concat(quote(column_name), ', ', column_name)), ') from sm_id where sm_id_pk = ? ;') into @sql "
+            + " from information_schema.columns where table_name = 'sm_id'";
+
+    private static final String SQL_GET_JSON_OBJECT_FOR_ONC_HISTORY = "select concat('select json_object(', group_concat( "
+            + " concat(quote(column_name), ', ', column_name)),') from ddp_onc_history_detail where onc_history_detail_id = ? ;')"
+            + " into @sql from information_schema.columns where table_name = 'ddp_onc_history_detail'";
+    
+    private static final String TABLE_PLACE_HOLDER = "$table";
+    private static final String PK_PLACE_HOLDER = "$pk";
+
+    private static final String SQL_DELETE_VALUES = "DELETE FROM $table WHERE $pk = ?";
+    private static final String SQL_INSERT_DELETED_VALUES = "INSERT INTO deleted_object SET original_table = ?, original_primary_key = ? ,"
+            + " data =?, deleted_by = ?,  deleted_date = ? ";
+
+    private static final String SQL_GET_DELETED_OBJECT_BY_ORIGINAL_ID_TABLE = "SELECT data FROM deleted_object WHERE original_table = ? "
+            + " AND original_primary_key = ? ";
+
+    public static boolean deletePatch(@NonNull String id, @NonNull String user,
```

Comment:
`id` is numeric, right?  If so, change this to an int/long (whichever is appropriate given the size of the column), use setInt/Long instead of setString, and have client callers do any string -> int conversion necessary.

Reply:
This id that comes from frontend is a String. I will change it both in the frontend code and here and make sure they match.

Reply:
It might not be worth changing it on the frontend--the important thing is that by the time it gets into this method, it's treated as an int (which could be a change to the signature, or could just be internal to the method to convert to an int/long.